src/Salt/SiteBundle/Controller/CommentsController.php

```diff
@@ -0,0 +1,215 @@
+    {
+        $em = $this->getDoctrine()->getManager();
+
+        $comment->setContent($request->request->get('content'));
+        $em->persist($comment);
+        $em->flush($comment);
+
+        $response = $this->apiResponse($comment);
+        return $response;
+    }
+
+    /**
+     * @Route("/comments/delete/{id}")
+     *
+     * @Method("DELETE")
+     *
+     * @Security("is_granted('comment')")
+     */
+    public function deleteAction(Comment $comment, UserInterface $user)
+    {
+        $em = $this->getDoctrine()->getManager();
+
+        $em->remove($comment);
+        $em->flush();
+
+        return new Response('Ok', 200);
+    }
+
+    /**
+     * @Route("/comments/{id}/upvote")
+     *
+     * @Method("POST")
+     *
+     * @Security("is_granted('comment')")
+     */
+    public function upvoteAction(Comment $comment, UserInterface $user)
+    {
+        $em = $this->getDoctrine()->getManager();
+
+        $commentUpvote = new CommentUpvote();
+        $commentUpvote->setComment($comment);
+        $commentUpvote->setUser($user);
+
+        $em->persist($commentUpvote);
+        $em->flush();
+
+        $response = $this->apiResponse($comment);
+        return $response;
+    }
+
+    /**
+     * @Route("/comments/{id}/upvote")
+     *
+     * @Method("DELETE")
+     *
+     * @Security("is_granted('comment')")
+     */
+    public function downvoteAction(Comment $comment, UserInterface $user)
+    {
+        $em = $this->getDoctrine()->getManager();
+
+        $commentUpvote = $em->getRepository('SaltSiteBundle:CommentUpvote')->findOneBy(
+            array('user' => $user, 'comment' => $comment)
+        );
+
+        if ($commentUpvote) {
+            $em->remove($commentUpvote);
+            $em->flush();
+
+            $response = $this->apiResponse($comment);
+            return $response;
+        }
+
+        return new Response('Item not found', 404);
+    }
+
+    private function existItem($itemId, $itemType)
+    {
+        if (filter_var($itemId, FILTER_VALIDATE_INT)) {
+            $em = $this->getDoctrine()->getManager();
+
+            switch ($itemType) {
+                case 'document':
+                    $item = $em->getRepository('CftfBundle:LsDoc')->findOneById($itemId);
+                    break;
+                case 'item':
+                    $item = $em->getRepository('CftfBundle:LsItem')->findOneById($itemId);
+                    break;
+                default:
+                    return false;
+            }
+
+            if ($item) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private function serialize($data)
+    {
+        return $this->get('jms_serializer')
+            ->serialize($data, 'json');
+    }
+
+    private function apiResponse($data, $statusCode = 200)
+    {
+        $json = $this->serialize($data);
+
+        return new Response($json, $statusCode, [
```

Comment:
This should probably create a `JsonResponse` instead of a `Response` (and then no need to set the additional header).